src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,94 @@
+package storage;
+
+import exception.DukeException;
+import task.Task;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+
+/**
+ * This Storage class is utilised to do both the reading and writing to
+ * persistent storage using the two primary methods saveFile and loadFile.
+ *
+ * @author Sai Ganesh Suresh
+ * @version v2.0
+ */
+public class Storage {
+
+    @SuppressWarnings("unchecked")
+    private static <T> T castToAnything(Object obj) {
+        return (T) obj;
+    }
+
+    private File file;
+
+    /**
+     * This Storage constructor is used to function is used to assign the different
+     * parameters required by the Storage methods.
+     *
+     * @param file This parameter holds the file to write to.
+     */
+    public Storage(File file) {
+        this.file = file;
+        this.file.getParentFile().mkdirs();
+    }
+
+    /**
+     * This saveFile method is used repeatedly throughout the other classes to save
+     * updates made to the TaskList to the persistent storage to ensure the user
+     * does not loose data due to sudden termination of the program.
+     *
+     * @param listOfTasks This parameter holds the updated TaskList of the user and
+     *                    used to save the updated TaskList.
+     * @throws DukeException This exception is thrown if there is not file at the
+     *                       given location to save to.
+     */
+    public void saveFile(ArrayList<Task> listOfTasks) throws DukeException {
+        try {
+            FileOutputStream fileOutputStream = new FileOutputStream(file);
+            ObjectOutputStream objectOutputStream = new ObjectOutputStream(fileOutputStream);
```

Comment:
This one again violates SLAP. IF the save file only gets services from other classes it does not need to do a lower-level work of creating those objects